sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -574,6 +578,99 @@ public Instant timestamp(DoFn<InputT, OutputT> doFn) {
               };
         }
         break;
+      case PTransformTranslation.SPLITTABLE_TRUNCATE_SIZED_RESTRICTION_URN:
+        if ((doFnSignature.truncateSizedRestriction() != null
+                && doFnSignature.truncateSizedRestriction().observesWindow())
+            || (doFnSignature.newTracker() != null && doFnSignature.newTracker().observesWindow())
+            || (doFnSignature.getSize() != null && doFnSignature.getSize().observesWindow())
+            || !sideInputMapping.isEmpty()) {
+          mainInputConsumer = this::processElementForWindowObservingTruncateSizedRestriction;
+          // OutputT == RestrictionT
+          this.processContext =
```

Comment:
Since this shares the same ProcessBundleContext as GetSize (except for the error context string), can we make this a class called SizedRestrictionWindowObservingProcessBundleContext that extends WindowObservingProcessBundleContext and is used in both places.

Ditto for the non window observing one as well.